rust/src/container.rs

```diff
@@ -418,6 +423,41 @@ pub fn container_encapsulate(args: Vec<String>) -> CxxResult<()> {
             .await
         })
     })?;
+
+    match &opt.compare_with_build {
+        Some(compare_with_build) => handle.block_on(async {
```

Comment:
This is more succinct as `if let Some(compare_with_build) = opt.compare_with_build.as_ref() {`

Reply:
Also, let's split the handling here into a separate function.